tools/pythonpkg/src/pyconnection.cpp

```diff
@@ -1146,7 +1146,10 @@ unique_ptr<DuckDBPyRelation> DuckDBPyConnection::ReadCSV(const py::object &name_
 			for (auto &kv : dtype_dict) {
 				shared_ptr<DuckDBPyType> sql_type;
 				if (!py::try_cast(kv.second, sql_type)) {
-					throw py::value_error("The types provided to 'dtype' have to be DuckDBPyType");
+					connection.context->RunFunctionInTransaction([&]() {
```

Comment:
This looks like we could start many different transactions when using this functionality. Is there a different way we can support custom types by e.g. postponing the binding of `USER` types?

Reply:
You mean due to the iteration? I could bring it in the function transaction, does that make sense?

Reply:
That would be better - but I wonder if this is the right place to do resolution of types in the first place. Can we not leave the type as a `USER` type here?

Reply:
All we are doing is converting the `sql_type` to a string after this anyway no?

Reply:
I think the original idea was to be able to have a nice error message from not matching with duckdbpytype. 

But I agree with you!